pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -581,6 +584,48 @@ func (qjm *XController) GetAggregatedResourcesPerGenericItem(cqj *arbv1.AppWrapp
 
 	return retVal
 }
+
+//Gets all objects owned by AW from API server, check user supplied status and set whole AW status
+func (qjm *XController) GetAllObjectsOwned(caw *arbv1.AppWrapper) arbv1.AppWrapperState {
+
+	// Get all pods and related resources
+	countCompletedItems := 0
+	countCompletionRequired := 0
+	for _, genericItem := range caw.Spec.AggrResources.GenericItems {
+		objectName := genericItem.GenericTemplate
+		var unstruct unstructured.Unstructured
+		unstruct.Object = make(map[string]interface{})
+		var blob interface{}
+		if err := jsons.Unmarshal(objectName.Raw, &blob); err != nil {
+			klog.Errorf("Error unmarshalling, err=%#v", err)
+		}
+
+		status := qjm.genericresources.GetGenericItemKindStatus(&genericItem, caw.Namespace)
+		if status {
+			countCompletedItems = countCompletedItems + 1
```

Comment:
Suggestion for performance reasons: if the stats is false the function can be returned with `caw.Status.State` since the condition will not change.  Don't think there is a need to continue counting?  Please provide your thoughts.  Thanks!

Probably a logging line of info-4 can added as well.

Reply:
Changed the logging level, thanks. 
I think we cannot do an early exit, we have to evaluate all items and then either set complete or RunningHoldCompletion. For instance AW with job and deployment, deployment never completes so AW will always be in Running state instead of being in RunningHoldCompletion state.